client/hwlib/src/collectors/hardware_info.rs

```diff
@@ -31,14 +31,8 @@ use crate::{
     models::devices::{Bios, Board, Chassis, Processor},
 };
 
-pub(crate) fn load_smbios_data(entry_filepath: &Path, table_filepath: &Path) -> Option<SMBiosData> {
-    match table_load_from_device(entry_filepath, table_filepath) {
-        Ok(data) => Some(data),
-        Err(e) => {
-            eprintln!("failed to load SMBIOS data: {}.", e);
-            None
-        }
-    }
+pub(crate) fn load_smbios_data(entry_filepath: &Path, table_filepath: &Path) -> Result<SMBiosData> {
+    table_load_from_device(entry_filepath, table_filepath)
```

Comment:
This function seems somewhat redundant now, would it be cleaner to move the content of `table_load_from_device` here? I'm not sure whether this is used elsewhere